ICAFIX/scripts/computeRecleanFeatures.m

```diff
@@ -0,0 +1,277 @@
+function computeRecleanFeatures(StudyFolder, ...
+                                Subject, ...
+                                fMRIListName, ...
+                                RunsXNumTimePoints, ...
+                                hp, ...
+                                Resolution, ...
+                                GrayOrdinateTemplateFile, ...
+                                CorticalParcellationFile, ...
+                                SubRegionParcellationFile, ...
+                                WMLabelFile, ...
+                                CSFLabelFile, VisualAreasFile, LanguageAreasFile, SubRegionsFile, NonGreyParcelsFile)
+
+HCPPIPEDIR = getenv('HCPPIPEDIR')
+
+GrayOrdinateTemplate=ciftiopen(GrayOrdinateTemplateFile,'wb_command');
+GrayOrdinateTemplate.cdata=single(zeros(size(GrayOrdinateTemplate.cdata))); % make a zero template
+FullGreyordinateLength = length(GrayOrdinateTemplate.cdata); 
+
+fMRINames = myreadtext(fMRIListName);
+RunsXNumTimePoints = str2double(RunsXNumTimePoints);
+ResolutionNum = str2double(Resolution);
+
+CorticalParcellation=ciftiopen(CorticalParcellationFile,'wb_command');
+VisualROI_CIFTI=ciftiopen(VisualAreasFile,'wb_command');
+VisualROI=VisualROI_CIFTI.cdata;
+NonVisualROI=1-VisualROI_CIFTI.cdata;
+
+LanguageROI_CIFTI=ciftiopen(LanguageAreasFile,'wb_command');
+LanguageROI=LanguageROI_CIFTI.cdata;
+NonLanguageROI=1-LanguageROI_CIFTI.cdata;
+
+SubRegionParcellation=ciftiopen(SubRegionParcellationFile,'wb_command');
+
+SubRegions=get_ROI_from_txt(SubRegionsFile); % SubRegionsFile HCPpipelines/global/config/SubRegions.ROI.txt
+
+NonGreyParcels=get_ROI_from_txt(NonGreyParcelsFile); % NonGreyParcelsFile HCPpipelines/global/config/NonGreyParcels.ROI.txt
+
+FinalSpatialSmoothingFWHM=4;
+
+% Calculate AdditionalSigma
+%AdditionalSigma = FinalSpatialSmoothingFWHM / (2 * (sqrt(2 * log(2))));
+%AdditionalSigmaString=num2str(AdditionalSigma, '%0.20f');
+
+subj=Subject;
+SubjFolderlist=[StudyFolder '/' subj];
+
+for j=1:length(fMRINames)
+    fMRIName=fMRINames{j};
+
+    sICA=load([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_mix']);
+    num_comps=size(sICA,2);
+
+    OrigFeatures=single(zeros(num_comps,186));
+    NewFeatures=single(zeros(num_comps,424));
+
+    if exist([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/Noise.txt'],'file')
+        sICA=load([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_mix']);
+        Noise=load([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/Noise.txt']);
```

Comment:
`Noise` looks like it isn't used?  Why are you specifically testing for the existence of `Noise.txt`?  If we wanted to replace the original classifier with this one, having it choose not to work because it didn't look like the old classifier ran yet would be somewhat silly.

Reply:
So you suggest we should assume the user is always aware of that the sICA+FIX has already been run?

Reply:
Why would the user need to know this detail?  The script should check what it needs to in order to work, but it doesn't appear to actually need the contents of Noise.txt, so I'm not sure what it should check for.  The `sICA` variable (could use a better name, like `sICAmix`) does get used, so maybe this `if` could instead check for the existence of the melodic_mix file?